src/Materials/shaderMaterial.ts

```diff
@@ -458,6 +506,51 @@ export class ShaderMaterial extends Material {
         return this;
     }
 
+    /**
+     * Set a uniform buffer in the shader
+     * @param name Define the name of the uniform as defined in the shader
+     * @param buffer Define the value to give to the uniform
+     * @return the material itself allowing "fluent" like uniform updates
+     */
+    public setUniformBuffer(name: string, buffer: UniformBuffer): ShaderMaterial {
+        if (this._options.uniformBuffers.indexOf(name) === -1) {
+            this._options.uniformBuffers.push(name);
+        }
+        this._uniformBuffers[name] = buffer;
+
+        return this;
+    }
+
+    /**
+     * Set a sampler in the shader
+     * @param name Define the name of the uniform as defined in the shader
+     * @param sampler Define the value to give to the uniform
+     * @return the material itself allowing "fluent" like uniform updates
+     */
+    public setSampler(name: string, sampler: Sampler): ShaderMaterial {
+        if (this._options.samplerObjects.indexOf(name) === -1) {
+            this._options.samplerObjects.push(name);
+        }
+        this._samplers[name] = sampler;
+
+        return this;
+    }
+
+    /**
+     * Set a storage buffer in the shader
+     * @param name Define the name of the storage buffer as defined in the shader
+     * @param buffer Define the value to give to the uniform
+     * @return the material itself allowing "fluent" like uniform updates
+     */
+     public setStorageBuffer(name: string, buffer: StorageBuffer): ShaderMaterial {

```

Comment:
Like for effects I wonder if it makes sense to move in a webgpu extension ?

Reply:
It probably makes less sense here :-)

Reply:
I could move the `setXXX` functions but there are some other code related to these new objects (storage texture, storage buffer, sampler) that are embedded into existing functions that I can't move. So I think it's better to keep the functions here?